iree/samples/dynamic_shapes/dynamic_shapes.ipynb

```diff
@@ -0,0 +1,555 @@
+            "      %3 = addi %arg1, %arg2 : i32 loc(#loc6)\n",
+            "      linalg.yield %3 : i32 loc(#loc6)\n",
+            "    } -> tensor<i32> loc(#loc6)\n",
+            "    return %2 : tensor<i32> loc(#loc0)\n",
+            "  } loc(#loc0)\n",
+            "} loc(#loc0)\n",
+            "#loc1 = loc(callsite(\"Const@__inference_add_one_7\"(\"/usr/local/lib/python3.7/dist-packages/tensorflow/python/framework/constant_op.py\":287:0) at callsite(\"/usr/local/lib/python3.7/dist-packages/tensorflow/python/framework/constant_op.py\":265:0 at callsite(\"<ipython-input-3-8994eb87ace0>\":22:0 at callsite(\"/usr/local/lib/python3.7/dist-packages/tensorflow/python/framework/func_graph.py\":982:0 at callsite(\"/usr/local/lib/python3.7/dist-packages/tensorflow/python/eager/function.py\":3971:0 at callsite(\"/usr/local/lib/python3.7/dist-packages/tensorflow/python/eager/def_function.py\":672:0 at callsite(\"/usr/local/lib/python3.7/dist-packages/tensorflow/python/framework/func_graph.py\":999:0 at callsite(\"/usr/local/lib/python3.7/dist-packages/tensorflow/python/eager/function.py\":3289:0 at callsite(\"/usr/local/lib/python3.7/dist-packages/tensorflow/python/eager/function.py\":3444:0 at \"/usr/local/lib/python3.7/dist-packages/tensorflow/python/eager/function.py\":3050:0))))))))))\n",
```

Comment:
it's probably worth trimming locations from this - since they have paths/line numbers/apparently randomized references to cells everyone will have different values and if someone else tries to update this notebook it'll just lead to noise.

Reply:
I'd like to trim them across all Colab notebooks, but I'm not sure if there's a good way to do that with the Python compiler bindings. As I understand it, there is probably a way to do that when calling the `iree-tf-import` tool directly.

Reply:
I've run mlir-opt / iree-opt with no args to sanitize IR dumps like these before, and could do that here... it's just a bit of extra code to find the path and issue the shell command when I'd rather stay in pure Python.

Reply:
I don't actually see `mlir-opt` or `iree-opt` in the packages, so not sure how I would trim these :/

Reply:
Let's ask stella how to do it - I'm pretty sure there's a flag to disable this already - we really don't want these in here, nor do we want them in artifacts we generate to check in as tests.

Reply:
@stellaraccident is there a convenient way to strip locations from imported MLIR?

```python
compiler_module = tfc.compile_module(DynamicShapesModule(), import_only=True)

# Print the imported MLIR to see how the compiler views this program.
print("Dynamic Shapes MLIR:\n```\n%s```\n" % compiler_module.decode("utf-8"))
```

produces

````
```
Dynamic Shapes MLIR:
#loc0 = loc(unknown)
#loc3 = loc(callsite("Sum@__inference_reduce_sum_1d_14"("/usr/local/lib/python3.7/dist-packages/tensorflow/python/ops/math_ops.py":2119:0) at callsite("/usr/local/lib/python3.7/dist-packages/tensorflow/python/util/dispatch.py":206:0 at callsite("<ipython-input-3-8ecf4caa1088>":10:0 at callsite("/usr/local/lib/python3.7/dist-packages/tensorflow/python/framework/func_graph.py":982:0 at callsite("/usr/local/lib/python3.7/dist-packages/tensorflow/python/eager/function.py":3971:0 at callsite("/usr/local/lib/python3.7/dist-packages/tensorflow/python/eager/def_function.py":672:0 at callsite("/usr/local/lib/python3.7/dist-packages/tensorflow/python/framework/func_graph.py":999:0 at callsite("/usr/local/lib/python3.7/dist-packages/tensorflow/python/eager/function.py":3289:0 at callsite("/usr/local/lib/python3.7/dist-packages/tensorflow/python/eager/function.py":3444:0 at "/usr/local/lib/python3.7/dist-packages/tensorflow/python/eager/function.py":3050:0))))))))))
#loc4 = loc(callsite("Sum@__inference_reduce_sum_2d_21"("/usr/local/lib/python3.7/dist-packages/tensorflow/python/ops/math_ops.py":2119:0) at callsite("/usr/local/lib/python3.7/dist-packages/tensorflow/python/util/dispatch.py":206:0 at callsite("<ipython-input-3-8ecf4caa1088>":16:0 at callsite("/usr/local/lib/python3.7/dist-packages/tensorflow/python/framework/func_graph.py":982:0 at callsite("/usr/local/lib/python3.7/dist-packages/tensorflow/python/eager/function.py":3971:0 at callsite("/usr/local/lib/python3.7/dist-packages/tensorflow/python/eager/def_function.py":672:0 at callsite("/usr/local/lib/python3.7/dist-packages/tensorflow/python/framework/func_graph.py":999:0 at callsite("/usr/local/lib/python3.7/dist-packages/tensorflow/python/eager/function.py":3289:0 at callsite("/usr/local/lib/python3.7/dist-packages/tensorflow/python/eager/function.py":3444:0 at "/usr/local/lib/python3.7/dist-packages/tensorflow/python/eager/function.py":3050:0))))))))))
#map0 = affine_map<(d0) -> ()>
#map1 = affine_map<(d0) -> (d0)>
#map2 = affine_map<(d0, d1) -> (d1, d0)>
#map3 = affine_map<(d0, d1) -> (d0)>
module  {
````

Reply:
I think the flag `--mlir-print-debuginfo=false` works.

For debugging, I have a shortcut that roundtrips through opt to simplify things (i.e. -mlir-elide-elementsattrs-if-larger= is usually pretty important).

There may be some flag triage/plumbing on some of the binaries.

Reply:
I'm also going to publish python -dev packages with the various opt binaries in it (with asserts enabled) for these things. Should allow copy-pasta templates to add to colab to pipe through various things for deeper development flows.

Reply:
```python
compiler_module = tfc.compile_module(
    DynamicShapesModule(), import_only=True, 
    output_mlir_debuginfo=False)
```
does what I want, thanks!

also tried
```python
import_extra_args=["--mlir-print-debuginfo=false"]
````
but that results in a duplicate flag error, per
https://github.com/google/iree/blob/b189c6e196c3428cd0a34aab3b61278c75881058/bindings/python/iree/compiler/core.py#L131
https://github.com/google/iree/blob/b189c6e196c3428cd0a34aab3b61278c75881058/bindings/python/iree/compiler/tf.py#L143-L147